core/src/main/java/com/scottlogic/datahelix/generator/core/generation/string/generators/RegexStringGenerator.java

```diff
@@ -121,40 +107,19 @@ public StringGenerator intersect(StringGenerator otherGenerator) {
             return new NoStringsStringGenerator("regex combination was contradictory");
         }
 
-        String mergedRepresentation = intersectRepresentation(
-            this.regexRepresentation,
-            ((RegexStringGenerator) otherGenerator).regexRepresentation);
+        RegexPattern intersectedPatterns = new AllRegexPatterns(
+            Arrays.asList(
+                this.regexPattern,
+                otherRegexGenerator.regexPattern));
 
-        return new RegexStringGenerator(merged, mergedRepresentation);
-    }
-
-    RegexStringGenerator union(RegexStringGenerator otherGenerator) {
-        Automaton b = otherGenerator.automaton;
-        Automaton merged = automaton.union(b);
-        String mergedRepresentation = unionRepresentation(
-            this.regexRepresentation,
-            otherGenerator.regexRepresentation
-        );
-        return new RegexStringGenerator(merged, mergedRepresentation);
+        return new RegexStringGenerator(merged, intersectedPatterns);
     }
 
     @Override
     public StringGenerator complement() {
         return new RegexStringGenerator(
             this.automaton.clone().complement().intersection(DEFAULT.automaton),
```

Comment:
If the defaults are already applied to this `automaton`, then taking the complement of it and intersecting it with `DEFAULT.automaton` will produce the empty set.

Reply:
Hmmm... maybe this has never worked?